dist/common/scripts/scylla_util.py

```diff
@@ -381,14 +384,13 @@ def __init__(self):
     @staticmethod
     def is_azure_instance():
         """Check if it's Azure instance via DNS lookup to metadata server."""
+        p = azure_instance()
         try:
-            addrlist = socket.getaddrinfo('metadata.azure.internal', 80)
-        except socket.gaierror:
+            is_imds = curl(p.META_DATA_BASE_URL + p.API_VERSION, headers = { "Metadata": "True" }, byte=False, timeout=3, max_retries=2, retry_interval=1)
```

Comment:
so this means on an on premise machine
this code will wait 2 + 2 x 3 seconds just to continue and say it's not Azure

in such case you can just go resolve 169.254.169.254